File: DSH.Main.Web/Boilerplate/src/modules/Models/Notification.js
define([], function () {
    var notification = function (context) {
        var self = this;
        self.id = ko.observable();
        self.senderId = ko.observable();
        self.recipientId = ko.observable();
        self.body = ko.observable();
        self.relevantPostId = ko.observable();
        self.relevantParentPostId = ko.observable();
        self.isRead = ko.observable();
        self.dateOfOrigin = ko.observable();
        self.senderDisplayName = ko.observable();
        self.senderPicUrl = ko.observable();
        self.url = ko.observable();
        self.details = ko.observable();

        context.listen("NOTIFICATIONS_READ", function () {
            if (self.isRead() == false) {
                $.ajax({
                    type: "GET",
                    url: "/Notification/MarkNotificationRead",
                    data: { notificationId: self.id },
                    success: function(result) {
                    }
                });
                self.isRead(true);
            }
        });
    };

    return notification;

});

File: DSH.Access/NotificationAccess/Model/Notification.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace DSH.Access.DataModels
{
    [Serializable]
    public class Notification
    {
        public int? Id { get; set; }
        public int? SenderId { get; set; }
        public int? RecipientId { get; set; }
        public string Body { get; set; }
        public int? RelevantPostId{get ;set;}
        public int? RelevantParentPostId { get; set; }
        public Boolean? IsRead { get; set; }
        public DateTime? DateOfOrigin { get; set; }
        public String SenderDisplayName { get; set; }
        public String SenderPicUrl { get; set; }
    }
}
